vault/identity_store_oidc_provider_test.go

```diff
@@ -1614,3 +1615,107 @@ func TestOIDC_Path_OIDC_Provider_List(t *testing.T) {
 	delete(expectedStrings, "test-provider2")
 	expectStrings(t, respListProvidersAfterDelete.Data["keys"].([]string), expectedStrings)
 }
+
+// TestOIDC_Path_OpenIDProviderConfig tests read operations for the
+// openid-configuration path
+func TestOIDC_Path_OpenIDProviderConfig(t *testing.T) {
+	c, _, _ := TestCoreUnsealed(t)
+	ctx := namespace.RootContext(nil)
+	storage := &logical.InmemStorage{}
+
+	// Create a test provider "test-provider"
+	resp, err := c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/provider/test-provider",
+		Operation: logical.CreateOperation,
+		Data:      map[string]interface{}{},
+		Storage:   storage,
+	})
+	expectSuccess(t, resp, err)
+
+	// Expect defaults from .well-known/openid-configuration
+	resp, err = c.identityStore.HandleRequest(ctx, &logical.Request{
+		Path:      "oidc/provider/test-provider/.well-known/openid-configuration",
+		Operation: logical.ReadOperation,
+		Storage:   storage,
+	})
+	expectSuccess(t, resp, err)
+
+	basePath := "/v1/identity/oidc/provider/test-provider"
+	expected := &providerDiscovery{
+		discovery: discovery{
+			Issuer:        basePath,
+			Keys:          basePath + "/.well-known/keys",
+			ResponseTypes: []string{"code"},
+			Subjects:      []string{"public"},
+			IDTokenAlgs:   supportedAlgs,
+		},
+		AuthorizationEndpoint: basePath + "/authorize",
+		TokenEndpoint:         basePath + "/token",
+		UserinfoEndpoint:      basePath + "/userinfo",
+	}
+	discoveryResp := &providerDiscovery{}
+	json.Unmarshal(resp.Data["http_raw_body"].([]byte), discoveryResp)
+	if diff := deep.Equal(expected, discoveryResp); diff != nil {
+		t.Fatal(diff)
+	}
+
+	// Update provider issuer config
```

Comment:
We could add an update to provider scopes to assert that `scopes_supported` is also updated in the discovery document.